components/collections/codepointtrie_builder/src/wasm.rs

```diff
@@ -32,64 +27,54 @@ where
         .to_owned(),
         format!("{}", std::mem::size_of::<T::ULE>() * 8),
     ];
-    let mut wasi_env = WasiState::new("list_to_ucptrie")
-        .stdin(Box::new(Pipe::new()))
-        .stdout(Box::new(Pipe::new()))
+    let mut store = Store::default();
+    let module = Module::new(&store, WASM_BYTES).expect("valid WASM");
```

Comment:
Question: it seems like a regression that we need to parse the bytes to a module every time. Before we did it in the lazy_static, such that the first call was slow but every subsequent call was fast. It appears that Wasmer 3 requires the store to be mutable. I wonder why? It seems like we should still be able to create a Module just once and then an Instance for each call.

Reply:
Actually does the store need to be the same for the Module and the Instance? Otherwise why would the API allow you to pass different arguments to both instead of just borrowing the store from the Module!

Reply:
The examples in https://github.com/wasmerio/wasmer/blob/v3.2.1/examples/hello_world.rs#L46-L71 also do the same.

Reply:
Yes but those hello world examples don't need to re-use the module. We should be able to parse the module once and not be required to re-parse it on every function call, just as we were able to do in Wasmer 2.0.

CC @syrusakbary for potential advice on updating from wasmer 2 to wasmer 3.